### ISSUE TITLE
rebootmgrctl not installed

### ISSUE THREAD

**Issue by tacerus** [openSUSE/os-update#18]
Hi,

with `REBOOT_CMD="rebootmgr"`, the following is observed on a Tumbleweed system:

```
Dec 26 14:27:07 mickey os-update[22093]: Since the last system boot core libraries or services have been updated.
Dec 26 14:27:07 mickey os-update[22093]: Reboot is suggested to ensure that your system benefits from these updates.
Dec 26 14:27:07 mickey os-update[3796]: ERROR: rebootmgrctl not installed
Dec 26 14:27:07 mickey systemd[1]: os-update.service: Main process exited, code=exited, status=1/FAILURE
Dec 26 14:27:07 mickey systemd[1]: os-update.service: Failed with result 'exit-code'.
Dec 26 14:27:07 mickey systemd[1]: Failed to start Update of system.
```

Inspecting the code, I do not see anything wrong with:

```
            "rebootmgr")
                if [ ! -x /usr/sbin/rebootmgrctl ]; then
                    log_error "ERROR: rebootmgrctl not installed"
                    exit 1
                fi
```

The file exists and is executable:

```
# ls -l /usr/sbin/rebootmgrctl
-rwxr-xr-x 1 root root 26816 Aug 21 09:48 /usr/sbin/rebootmgrctl
# rpm -qf /usr/sbin/rebootmgrctl
rebootmgr-2.1-1.1.x86_64
```

In a shell, the test passes seemingly as expected:

```
# [ ! -x /usr/sbin/rebootmgrctl ] ; echo $?
1
```

The file does not seem to have been "modified" during the last update either:

```
# stat /usr/sbin/rebootmgrctl 
  File: /usr/sbin/rebootmgrctl
  Size: 26816           Blocks: 56         IO Block: 4096   regular file
Device: 254,1   Inode: 92269       Links: 1
Access: (0755/-rwxr-xr-x)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 2023-08-21 09:48:29.000000000 +0000
Modify: 2023-08-21 09:48:29.000000000 +0000
Change: 2023-11-19 22:12:14.872569861 +0000
 Birth: 2023-11-19 22:12:14.869236399 +0000
# date
Tue Dec 26 14:34:34 UTC 2023
```

This is with `os-update-1.14-1.1.noarch` on a Tumbleweed snapshot as of today.

**Comment by thkukuk** [comment]
This bug is fixed in 1.14 but was in older versions and is working fine on many of our servers:
```
+ case "$REBOOT_CMD" in
+ '[' '!' -x /usr/sbin/rebootmgrctl ']'
+ /usr/sbin/rebootmgrctl is-active -q
+ /usr/sbin/rebootmgrctl reboot
+ exit 0
```
So either you are not using os-update 1.14 or you have some restrictive systemd, apparmor, SELinux or other security rules installed which prevent the os-update process to access the command.
To me it looks like as if you haven't done an update for a longer time (there were no updates released in the last days which trigger a reboot) and the update was still run with os-update 1.13.

**Comment by tacerus** [comment]
Indeed, it seems 1.14 got installed along with the problematic update:

```
2023-12-26 14:24:27|install|os-update|1.14-1.1|noarch||repo-oss|fbdc18982c0ad22fa8410d8a427d0fd3aa014a5f2503b2a81253e64bdae01f5c760be5703a15c582a50eba3818445b09ac5a164d9f1d1df2027a8ffc08b75a3d|
```

I did not realize the code was different in the previous version.
Then I'm sure it will run just fine next time.

Thank you!
